Tests/TuistKitTests/Generator/WorkspaceStructureGeneratorTests.swift

```diff
@@ -0,0 +1,344 @@
+
+        // Then
+        XCTAssertEqual(structure.contents, [
+            .group("Modules", "/path/to/workspace/Modules", [
+                .project("/path/to/workspace/Modules/A"),
+                .group("A", "/path/to/workspace/Modules/A", [
+                    .file("/path/to/workspace/Modules/A/README.md"),
+                ]),
+            ]),
+        ])
+    }
+
+    // MARK: - Helpers
+
+    @discardableResult
+    func createFolders(_ folders: [String]) throws -> [AbsolutePath] {
+        let paths = folders.map { AbsolutePath($0) }
+        try paths.forEach {
+            try fileHandler.createFolder($0)
+        }
+        return paths
+    }
+
+    @discardableResult
+    func createFiles(_ files: [String]) throws -> [AbsolutePath] {
+        let paths = files.map { AbsolutePath($0) }
+        try paths.forEach {
+            try fileHandler.touch($0)
+        }
+        return paths
+    }
+
+    class InMemoryFileHandler: FileHandling {
+        private enum Node {
+            case file
+            case folder
+        }
+
+        private var cache: [AbsolutePath: Node] = [:]
+
+        var currentPath: AbsolutePath = "/"
+
+        func replace(_: AbsolutePath, with _: AbsolutePath) throws {}
+
+        func exists(_ path: AbsolutePath) -> Bool {
+            return cache[path] != nil
+        }
+
+        func copy(from _: AbsolutePath, to _: AbsolutePath) throws {}
+
+        func readTextFile(_: AbsolutePath) throws -> String {
+            return ""
+        }
+
+        func inTemporaryDirectory(_: (AbsolutePath) throws -> Void) throws {}
+
+        func glob(_: AbsolutePath, glob _: String) -> [AbsolutePath] {
+            return []
+        }
+
+        func createFolder(_ path: AbsolutePath) throws {
+            var pathSoFar = AbsolutePath("/")
+            for component in path.components.dropFirst() {
+                pathSoFar = pathSoFar.appending(component: component)
+                cache[pathSoFar] = .folder
+            }
+        }
+
+        func delete(_ path: AbsolutePath) throws {
+            cache.removeValue(forKey: path)
+        }
+
+        func isFolder(_ path: AbsolutePath) -> Bool {
+            return cache[path] == .folder
+        }
+
+        func touch(_ path: AbsolutePath) throws {
+            let parent = path.parentDirectory
+            try createFolder(parent)
+            cache[path] = .file
+        }
+
+        func ls(_: AbsolutePath) throws -> [AbsolutePath] {
+            return []
+        }
+    }
+}
+
+extension WorkspaceStructure.Element {
+    static func group(_ name: String,
+                      _ path: AbsolutePath,
+                      _ contents: [WorkspaceStructure.Element]) -> WorkspaceStructure.Element {
+        return .group(name: name, path: path, contents: contents)
+    }
+
+    static func project(_ path: AbsolutePath) -> WorkspaceStructure.Element {
+        return .project(path: path)
```

Comment:
what are the point of these functions?

Reply:
purely to minify the test results comparisons - maybe overkill :/

e.g.

```swift
XCTAssertEqual(structure.contents, [
            .group(name: "Modules", path: "/path/to/workspace/Modules", contents: [
                .project(path: "/path/to/workspace/Modules/A"),
                .project(path: "/path/to/workspace/Modules/B"),
                .group(name: "Sub", path: "/path/to/workspace/Modules/Sub", contents: [
                    .project(path: "/path/to/workspace/Modules/Sub/C"),
                    .project(path: "/path/to/workspace/Modules/Sub/D"),
                ]),
            ]),
        ])
```

vs

```swift
XCTAssertEqual(structure.contents, [
            .group("Modules", "/path/to/workspace/Modules", [
                .project("/path/to/workspace/Modules/A"),
                .project("/path/to/workspace/Modules/B"),
                .group("Sub", "/path/to/workspace/Modules/Sub", [
                    .project("/path/to/workspace/Modules/Sub/C"),
                    .project("/path/to/workspace/Modules/Sub/D"),
                ]),
            ]),
        ])
```